repose-aggregator/components/filters/forwarded-proto/src/main/scala/org/openrepose/filters/forwardedproto/ForwardedProtoFilter.scala

```diff
@@ -0,0 +1,41 @@
+package org.openrepose.filters.forwardedproto
+
+import javax.servlet._
+import javax.servlet.http.HttpServletRequest
+
+import com.rackspace.httpdelegation._
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.utils.servlet.http.MutableHttpServletRequest
+
+/**
+ * The sole purpose of this filter is to add the X-Forwarded-Proto header to a request with a value which
+ * corresponds to the protocol of the request (e.g., http or https).
+ */
+class ForwardedProtoFilter extends Filter with HttpDelegationManager with LazyLogging {
+
+  private final val X_FORWARDED_PROTO = "X-Forwarded-Proto"
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("ForwardedProto filter initialized")
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    val httpServletRequest = servletRequest.asInstanceOf[HttpServletRequest]
+
+    if (Option(httpServletRequest.getHeader(X_FORWARDED_PROTO)).isEmpty) {
+      logger.debug(s"Adding the $X_FORWARDED_PROTO header")
+
+      val mutableHttpServletRequest = MutableHttpServletRequest.wrap(httpServletRequest)
+      mutableHttpServletRequest.addHeader(X_FORWARDED_PROTO, servletRequest.getProtocol)
```

Comment:
Dunno. So what we get from that call is detailed here: http://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getProtocol()

X-Forwarded-Proto is an extension header, so I don't believe that there is a formal specification for it. There may be standards for it, though. It appears that the "Forwarded" header has been specified in http://tools.ietf.org/html/rfc7239 and is meant as a substitute for the X-Forwarded-Proto header.

I don't expect the value overly matters as long as the origin service knows what to expect. With that said, I'm fine changing the header name, value, or both.